### PR TITLE
test: adds scheduled smoke-tests

### PR DESCRIPTION
**Pull request by AriPerkkio**
**What**:

See discussion from https://github.com/testing-library/eslint-plugin-jest-dom/issues/97.
This PR will add weekly scheduled smoke tests to be run on every Sunday at 00:00. These tests will verify that rules of `recommended` set do not cause linter crashes.

The test set contains 646 repositories. My initial `repositories.json` contained ~150 repositories so I included ~500 more by querying libraries.io API. These repositories had `@testing-library/react` as dependency.

Contents of each of these repositories are run against this ESLint plugin's **production build**, the `./dist` directory. 

Example run on CI: https://github.com/AriPerkkio/eslint-plugin-jest-dom/actions/runs/390036475

**Why**:

Unit tests cover only cases which are expected. The AST produced by Javascript and Typescript can cause very unexpected conditions. These often lead into null pointers and can crash the linter. 

ESLint rules should never crash. A crashing rule causes linter to lose all valid linting problems. 

The current master branch seems to contain two bugs which were found while setting up this PR:

<details>
  <summary>Crash reports</summary>

### Rule: prefer-to-have-style
Message: `Cannot read property 'replace' of undefined` Occurred while linting <text>:74
Path: `roginfarrer/react-collapsed/src/__tests__/index.test.tsx`
[Link](https://github.com/roginfarrer/react-collapsed/blob/HEAD/src/__tests__/index.test.tsx#L74)

```ts
  );
  const collapse = getByTestId('collapse');
  expect(collapse).toHaveAttribute('id');
  expect(collapse).toHaveAttribute('aria-hidden', 'false');
  expect(collapse.style).not.toContain(
    expect.objectContaining({
      display: 'none',
      height: '0px',
    })
  );
```

### Rule: prefer-to-have-style
Message: `Cannot read property 'range' of undefined` Occurred while linting <text>:37
Path: `FormidableLabs/spectacle/src/components/deck/deck.test.js`
[Link](https://github.com/FormidableLabs/spectacle/blob/HEAD/src/components/deck/deck.test.js#L37)

```js
    );
    expect(tree.find('AnimatedDeckDiv').props().style).toHaveProperty(
      'transform'
    );
    expect(tree.find('AnimatedDeckDiv').props().style).not.toHaveProperty(
      'opacity'
    );
  });

  it('should pass the transition effect to children slides', () => {
```

</details>

**How**:

Smoke tests are run on CI. The Github action will build the production build, link local dependency to point at this directory and scan the given repositories. The process will exit with error code in case of errors.
The job can also be ran manually, `workflow_dispatch`.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [x] Ready to be merged
      

As discussed on the issue earlier, it would be great if Github Issues could be created automatically. I have not implemented this but I've added a feature which allows such integration.
See the `onComplete` property of [configuration file](https://github.com/AriPerkkio/eslint-remote-tester#eslint-remote-testerconfigjs).

I had to push this commit with `--no-verify` option. The `kcd-scripts` froze on the test step, `kcd-scripts test --findRelatedTests`.

All comments are welcome!

### CODE REVIEW

File: smoke-test/eslint-remote-tester.config.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            sourceType: 'module',
+            ecmaFeatures: {
+                jsx: true,
+            },
+        },
+        plugins: [
+            'jest-dom',
+        ],
+        extends: [
+            'plugin:jest-dom/recommended',
```
Comment (benmonro) [inline smoke-test/eslint-remote-tester.config.js:67]: @AriPerkkio is it possible to run non recommended rules as well?  For example, we recently added a new rule `prefer-in-document` which we want to vet before putting in the recommended config.

**Comment by benmonro** [comment]
this is so cool! 🔥 

only one question, how did you generate the list of repositories for it to test against?

**Comment by benmonro** [comment]
@all-contributors please add @AriPerkkio for code, bugs, tests

**Comment by benmonro** [comment]
@AriPerkkio is it possible to have a manual trigger?  if so how does that work?

**Comment by AriPerkkio** [comment]
> is it possible to have a manual trigger? if so how does that work?

The `workflow_dispatch` is manual trigger in Github Actions. https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#manual-events

```yml
on:
  schedule:
    - cron: '0 0 * * SUN'
  workflow_dispatch:

```
Note that the syntax above requires that additional `:` since `schedule` has properties. It looks weird but is valid yml. 

The button for triggering manual event is found from Actions tab:

image

File: smoke-test/eslint-remote-tester.config.js (hunk 1 above)
Comment (AriPerkkio) [reply smoke-test/eslint-remote-tester.config.js:67]: `prefer-in-document` is now added.

Comment (AriPerkkio) [reply smoke-test/eslint-remote-tester.config.js:67]: @benmonro FYI, just realized this after the merge. I'm not sure but this should probably have been prefixed with plugin name:

```
'jest-dom/prefer-in-document': 'error'
```

It's currently now showing any errors but the rule might not have been loaded by ESLint.

Comment (benmonro) [reply smoke-test/eslint-remote-tester.config.js:67]: Oh good catch. Actually I might add another config with all the rules set to error anyway so we can just switch to that

**Pull request merged**